src/main/java/io/confluent/connect/elasticsearch/jest/JestElasticsearchClient.java

```diff
@@ -402,8 +404,11 @@ private void createIndex(String index, CreateIndex createIndex) throws ConnectEx
       JestResult result = client.execute(createIndex);
       log.debug("Received response for request to create index '{}'", index);
       if (!result.isSucceeded()) {
+        boolean exists = result.getErrorMessage().contains(RESOURCE_ALREADY_EXISTS_EXCEPTION)
+            || indexExists(index);
+
         // Check if index was created by another client
-        if (!indexExists(index)) {
+        if (!exists) {
           String msg =
               result.getErrorMessage() != null ? ": " + result.getErrorMessage() : "";
```

Comment:
Does this logic check if the index was created by another client (like the comment says)? It seems to just check if it existed, not if it was by a different client

Reply:
Q: Wouldn't it make sense to check `indexExists(index)` before executing the action to create index rather than the other way around? Then after creation of index, if it still failed, we can add debug logs to understand the real root cause

Reply:
https://github.com/confluentinc/kafka-connect-elasticsearch/pull/449/files#diff-784dd8ae0d081fd9ae1b2ecc191392a2L371

Reply:
the logic flow is a bit confusing but we do check beforehand